### DISCUSSION TITLE
jpg to avif conversion is broken

### DISCUSSION THREAD

**Discussion by basings** [ImageMagick/ImageMagick#6679] [Contribute]
### ImageMagick version

ImageMagick 7.1.0-45 Q16-HDRI x86_64 20319

### Operating system

Linux

### Operating system, version and so on

EndeavourOS, up to date

### Description

When I convert following jpg image to avif, the final avif version is not properly shown on Android 12. It's shown as a 180 MP image (8211 × 22542⁩) and doesn't get rendered. This is just an example image, it happens with more images as well. At first, I thought it's a bug of aves (https://github.com/deckerst/aves/issues/296) (my image gallery on android) because it's shown without problems in eye of gnome and can be opened by gimp.

used command: `convert DSC00519.jpg DSC00519.avif`

link to final avif image is attached in a zip folder. original jpg image is attached as well.

### Steps to Reproduce

1. download example image
2. run `convert DSC00519.jpg DSC00519.avif`
3. transfer photo to android and open image in a gallery app

### Images

![DSC00519](https://user-images.githubusercontent.com/92867705/182685452-dea709e0-49b3-47d0-a220-5222b62c0f32.JPG)
[DSC00519.zip](https://github.com/ImageMagick/ImageMagick/files/9254191/DSC00519.zip)

**Comment by urban-warrior** [comment]
ImageMagick converts your image to a HEIF image with AVIF encoding.  Does Android support HEIF containers?

**Comment by pikatchu2k3** [comment]
@urban-warrior 
I wanted to confirm this bug.
Most of the converted files from jpg to avif are not shown in Android (Simple Gallery, Aves, Google Fotos). Even removing the Exif data doesn´t solve the problem.
On Manjaro I can see all the files.

Tested on Android 13, which is supposed to support HEIF converted files depending on this android document under 5.1.16.
https://source.android.com/docs/compatibility/13/android-13-cdd

Used conversion format was:
magick mogrify -format avif -depth 10 -define heic:speed=2 *.jpg

**Comment by urban-warrior** [comment]
We leverage the libheif delegate library to convert to the AVIF format.  The HEIF distribution includes a `heif-convert` tool.  Create a AVIF with it.  If Android fails to read that image, the problem is likely with Android or libheif.  If Android can read the AVIF image, the issue likely is within ImageMagick.  Let us know the results of this test.

**Comment by pikatchu2k3** [comment]
@urban-warrior 
I tried successfully the heif-enc process with the command `heif-enc -q 40  a.jpeg -o a.avif`
Then I took the same jpeg Image and converted it with `magick mogrify -format avif -depth 10 -define heic:speed=2 a.jpeg `
Only the file created with libheif is shown on Android Fotos.

**Comment by agguser** [comment]
Any update or workaround?
